src/matchers/image.rs

```diff
@@ -154,7 +156,7 @@ fn get_ftyp(buf: &[u8]) -> Option<(&[u8], &[u8], impl Iterator<Item = &[u8]>)> {
 
     let major = &buf[8..12];
     let minor = &buf[12..16];
-    let compatible = (16..ftyp_length).step_by(4).map(move |i| &buf[i..i + 4]);
+    let compatible = buf[16..].chunks_exact(4);
```

Comment:
This elides the bounds check in `&[i..i + 4]`, which the compiler rightfully put there since the size of `buf` might not be divisible by 4, and in that case we could go out of bounds.

[`chunk_exact`](https://doc.rust-lang.org/stable/std/primitive.slice.html#method.chunks_exact) fixes the issue and removes the bounds check